docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Connecting to a service
+sidebar_position: 1
+---
+
+This tutorial will show you how you can connect your Go application to a mongodb service.
+
+Building on top of the [Go quickstart guide](../quickstart/go.md), we will extend the application to check if it is connected to a mongodb service.
+
+### Pre-requisites
+1. Install Service Binding Operator on your cluster.  See [installation instruction](https://operatorhub.io/operator/service-binding-operator).
+
+2. Install Percona Server Mongodb operator on your cluster. See [installation instruction](https://operatorhub.io/operator/percona-server-mongodb-operator).
+:::note
+The operator will be installed in a new namespace called "my-percona-server-mongodb-operator" and will be usable from that namespace only.
+:::
+3. Create a mongodb service.
+
+import CreateMongodbService from './_create-mongodb-service.mdx';
+
+<CreateMongodbService/>
+
+### Implement the code logic
+:::note
+If you're already running `odo dev` in a terminal, exit it and start afresh.
+:::
+
+The new code is simple. We obtain the connection information (username, password, and host) from the environment, and use it to connect to the mongodb service and ping it. 
+
+Replace the content of your `main.go` with the following content:
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// Connection URI
+var (
+	USERNAME = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_USER")
+	PASSWORD = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_PASSWORD")
+	HOST     = os.Getenv("PERCONASERVERMONGODB_HOST")
+	uri      = fmt.Sprintf("mongodb://%s:%s@%s:27017/?maxPoolSize=20&w=majority", USERNAME, PASSWORD, HOST)
+)
+
+func main() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe("0.0.0.0:8080", nil)
```

Comment:
shouldn't port be dynamic / so we can change it within devfile deploy?

Reply:
I've simply extended the quickstart guide, but perhaps you're right. I'll look into that.

Reply:
I am not sure how to address this comment. The default port in the devfile.yaml is 8080, and it works for this guide. We do not expose the port information in environment in any way that I could obtain it. I could modify main.go to read the devfile and obtain port from it, but it doesn't seem worth the effort to me. I think I'd keep things as is for now,  unless you have a better solution.